### PR TITLE
Add gofumpt

### PR DESCRIPTION
**Pull request by tomlazar**
Basically the same as gofmt, just a little more strict.

### CODE REVIEW

File: linters/gofumpt/gofumpt.test.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { linterCheckTest, linterFmtTest } from "tests";
+
+// gofumpt generates a failure on an empty file
+linterCheckTest({ linterName: "gofumpt", namedTestPrefixes: ["empty"] });
```
Comment (TylerJang27) [inline linters/gofumpt/gofumpt.test.ts:4]: Thanks for calling this out. We may move in the future to adding `allow_empty_files: false` to both gofumpt and gofmt, but this should be fine for now.

File: package-lock.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "plugins",
+  "name": "trunk-plugins",
```
Comment (TylerJang27) [inline package-lock.json:2]: Can you back this file change out? Was there something you ran into that brought this up?

**Review by TylerJang27 — APPROVED** [review]
Hi there! Thanks for using trunk and for making this contribution! The setup for gofumpt looks great, with just a couple minor comments.

Once you've addressed those feel free to update your PR with main and merge! Thanks again!

File: package-lock.json (hunk 2 above)
Comment (tomlazar) [reply package-lock.json:2]: nothing specific, must have happened since I renamed my fork as `trunk-plugins`

**Comment by tomlazar** [comment]
@TylerJang27 should be all good to go!

**Comment by TylerJang27** [comment]
@tomlazar Thanks again! We probably won't get another release out for a week or so, but if you need `gofumpt` sooner, you can change the plugin source in your trunk.yaml to look like:

```yaml
plugins:
  sources:
    - id: trunk
      ref: 19526b8797860900fb9680233943072b78c9f356
      uri: https://github.com/trunk-io/plugins
```

**Pull request merged**